### PR TITLE
Remove allocations in `jtprod!`

### PR DESCRIPTION
**Pull request by tmigot**
#74

### CODE REVIEW

File: src/slack-model.jl [hunk 1]
```diff
@@ -446,6 +446,47 @@ function NLPModels.jprod_nln!(
...
+  nlp::SlackModels,
+  x::AbstractVector{T},
+  v::AbstractVector,
+  jtv::AbstractVector,
+) where {T}
+  @lencheck nlp.meta.nvar x jtv
+  @lencheck nlp.meta.ncon v
+  n = nlp.model.meta.nvar
+  @views begin
+    jtprod!(nlp.model, x[1:n], v, jtv[1:n])
```
Comment (geoffroyleconte) [inline src/slack-model.jl:459]: is @views begin really necessary? I feel like you only need it for L459?

Comment (tmigot) [reply src/slack-model.jl]: Nice catch, thanks! I updated it.

**Review by geoffroyleconte — APPROVED** [review]

File: src/slack-model.jl [hunk 2]
```diff
@@ -446,6 +446,46 @@ function NLPModels.jprod_nln!(
...
+  nlp::SlackModels,
+  x::AbstractVector{T},
+  v::AbstractVector,
+  jtv::AbstractVector,
+) where {T}
+  @lencheck nlp.meta.nvar x jtv
+  @lencheck nlp.meta.ncon v
+  n = nlp.model.meta.nvar
+  @views jtprod!(nlp.model, x[1:n], v, jtv[1:n])
+  jtv[n + 1:end] .= zero(T) # Just for init purpose
```
Comment (dpo) [inline src/slack-model.jl:459]: It seems like this line could be skipped.

Comment (tmigot) [reply src/slack-model.jl:459]: Thanks for the suggestion! I open a new PR to fix this.
https://github.com/JuliaSmoothOptimizers/NLPModelsModifiers.jl/pull/84

**Pull request merged**